server-spi/src/main/java/org/keycloak/models/RealmProvider.java

```diff
@@ -216,11 +216,6 @@ default GroupModel getGroupById(String id, RealmModel realm) {
         return getGroupById(realm, id);
     }
 
-    /**
-     * @deprecated Use the corresponding method from {@link GroupProvider}. */
-    @Override
-    List<GroupModel> getGroups(RealmModel realm);
```

Comment:
I think we should not remove these methods, they are here so that users can see they are using deprecated methods.